src/control/server/config/server_test.go

```diff
@@ -557,32 +605,79 @@ func TestServerConfig_Parsing(t *testing.T) {
 			},
 			expValidateErr: errors.New("bdev_list contains duplicate pci"),
 		},
+		"legacy storage; empty bdev_list": {
+			legacyStorage: true,
+			expCheck: func(c *Server) error {
+				nr := len(c.Engines[0].Storage.Tiers)
+				if nr != 1 {
+					return errors.Errorf("want %d storage tiers, got %d", 1, nr)
+				}
+				return nil
+			},
+		},
+		"legacy storage; no bdev_list": {
+			legacyStorage: true,
+			inTxt:         "  bdev_list: []",
+			outTxt:        "",
+			expCheck: func(c *Server) error {
+				nr := len(c.Engines[0].Storage.Tiers)
+				if nr != 1 {
+					return errors.Errorf("want %d storage tiers, got %d", 1, nr)
+				}
+				return nil
+			},
+		},
+		"legacy storage; no bdev_class": {
+			legacyStorage: true,
+			inTxt:         "  bdev_class: nvme",
+			outTxt:        "",
+			expCheck: func(c *Server) error {
+				nr := len(c.Engines[0].Storage.Tiers)
+				if nr != 1 {
+					return errors.Errorf("want %d storage tiers, got %d", 1, nr)
+				}
+				return nil
+			},
+		},
+		"legacy storage; non-empty bdev_list": {
+			legacyStorage: true,
+			inTxt:         "  bdev_list: []",
+			outTxt:        "  bdev_list: [0000:80:00.0]",
+			expCheck: func(c *Server) error {
+				nr := len(c.Engines[0].Storage.Tiers)
+				if nr != 2 {
+					return errors.Errorf("want %d storage tiers, got %d", 2, nr)
+				}
+				return nil
+			},
+		},
 	} {
 		t.Run(name, func(t *testing.T) {
 			log, buf := logging.NewTestLogger(t.Name())
 			defer ShowBufferOnFailure(t, buf)
 
-			if tt.extraConfig == nil {
-				tt.extraConfig = noopExtra
-			}
-
 			testDir, cleanup := CreateTestDir(t)
 			defer cleanup()
 
-			// First, load a config based on the server config with all options uncommented.
-			testFile := filepath.Join(testDir, sConfigUncomment)
-			uncommentServerConfig(t, testFile)
-
-			replaceFile(t, testFile, tt.inTxt, tt.outTxt)
+			if tt.extraConfig == nil {
+				tt.extraConfig = noopExtra
+			}
 
-			config, errParse := mockConfigFromFile(t, testFile)
+			config, errParse := loadFromFile(t, testDir, tt.inTxt, tt.outTxt, tt.legacyStorage)
 			CmpErr(t, tt.expParseErr, errParse)
 			if tt.expParseErr != nil {
 				return
 			}
 			config = tt.extraConfig(config)
+			log.Debugf("%+v", config)
```

Comment:
this has snuck in un-intentionally and I will remove in subsequent PR, it will only print if the test fails so it's harmless